examples/experimental/dagster-airlift/dagster_airlift/core/serialization/serialized_data.py

```diff
@@ -113,29 +62,42 @@ class SerializedDagData:
     all_asset_keys_in_tasks: AbstractSet[AssetKey]
 
 
+@whitelist_for_serdes
+@record
+class KeyScopedDataItem:
+    asset_key: AssetKey
+    data: "SerializedAssetKeyScopedAirflowData"
```

Comment:
I'd prefer to use bare tuples. Note sure why this is necessary

Appeared to get coerced into a list in the machinery but did not investigate particularly thoroughly:

```
____________________________________________________________________________ test_cached_loading _____________________________________________________________________________

    def test_cached_loading() -> None:
        """Test cached loading behavior."""
        a = AssetKey("a")
        spec = AssetSpec(
            key=a,
            metadata=metadata_for_task_mapping(task_id="task", dag_id="dag"),
        )
        instance = make_instance({"dag": ["task"]})
        passed_in_defs = Definitions(assets=[spec])

        defs = build_defs_from_airflow_instance(airflow_instance=instance, defs=passed_in_defs)
        assert defs.assets
        assert len(list(defs.assets)) == 2
        assert {
            key for assets_def in defs.assets for key in cast(AssetsDefinition, assets_def).keys
        } == {a, AssetKey(["airflow_instance", "dag", "dag"])}
        assert len(defs.metadata) == 1
        assert "dagster-airlift/source/test_instance" in defs.metadata
        assert isinstance(defs.metadata["dagster-airlift/source/test_instance"].value, str)
>       assert isinstance(
            deserialize_value(defs.metadata["dagster-airlift/source/test_instance"].value),
            AirflowDefinitionsData,
        )

examples/experimental/dagster-airlift/dagster_airlift_tests/unit_tests/core_tests/test_load_defs.py:349:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
python_modules/dagster/dagster/_serdes/serdes.py:1118: in deserialize_value
    return deserialize_values([val], as_type, whitelist_map)[0]
python_modules/dagster/dagster/_serdes/serdes.py:1161: in deserialize_values
    unpacked_value = seven.json.loads(
../../../.pyenv/versions/3.11.9/lib/python3.11/json/__init__.py:359: in loads
    return cls(**kw).decode(s)
../../../.pyenv/versions/3.11.9/lib/python3.11/json/decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
../../../.pyenv/versions/3.11.9/lib/python3.11/json/decoder.py:353: in raw_decode
    obj, end = self.scan_once(s, idx)
python_modules/dagster/dagster/_serdes/serdes.py:1198: in _unpack_object
    return deserializer.unpack(val, whitelist_map, context)
python_modules/dagster/dagster/_serdes/serdes.py:612: in unpack
    value = self.handle_unpack_error(exc, context, unpacked_dict)
python_modules/dagster/dagster/_serdes/serdes.py:635: in handle_unpack_error
    raise exc
python_modules/dagster/dagster/_serdes/serdes.py:610: in unpack
    return self.klass(**unpacked)
<string>:7: in __checked_new__
    ???
python_modules/dagster/dagster/_check/functions.py:740: in list_param
    return _check_iterable_items(obj, of_type, "list")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

obj_iter = [[AssetKey(['a']), SerializedAssetKeyScopedAirflowData(additional_metadata={'Computed in Task ID': 'task', 'Dag ID': '...ummy.domain/dags/dag'), 'Task Info (raw)': JsonMetadataValue(data={})}, additional_tags={'dagster/kind/airflow': ''})]]
of_type = typing.Tuple, collection_name = 'list'

    def _check_iterable_items(
        obj_iter: T_Iterable, of_type: TypeOrTupleOfTypes, collection_name: str = "iterable"
    ) -> T_Iterable:
        for obj in obj_iter:
            if not isinstance(obj, of_type):
                if isinstance(obj, type):
                    additional_message = (
                        " Did you pass a class where you were expecting an instance of the class?"
                    )
                else:
                    additional_message = ""
>               raise CheckError(
                    f"Member of {collection_name} mismatches type. Expected {of_type}. Got"
                    f" {obj!r} of type {type(obj)}.{additional_message}"
                )
E               dagster._check.functions.CheckError: Member of list mismatches type. Expected typing.Tuple. Got [AssetKey(['a']), SerializedAssetKeyScopedAirflowData(additional_metadata={'Computed in Task ID': 'task', 'Dag ID': 'dag', 'Link to DAG': UrlMetadataValue(url='http://dummy.domain/dags/dag'), 'Task Info (raw)': JsonMetadataValue(data={})}, additional_tags={'dagster/kind/airflow': ''})] of type <class 'list'>.
```

Reply:
I experienced similar behavior when mucking around with initial versions of the task mapping. I think it might be a json thing, since json doesn't really support tuples top level? They just get coerced via iterability or something?